contracts/src/StakeTable.sol

```diff
@@ -166,32 +164,37 @@ contract StakeTable is AbstractStakeTable {
         return epoch;
     }
 
-    /// @notice get the next available epoch based and queue size
-    /// @param firstAvailableEpoch The first available epoch (for registration or exit)
-    /// @param numPending The number of pending exits or registrations
-    /// @return epoch The next available epoch
-    /// @return queueSize The size of the queue in that epoch
+    /// @notice Add a registration
     /// TODO handle overflow when max uint64 is reached
-    function nextAvailableEpoch(uint64 firstAvailableEpoch, uint64 numPending)
-        external
-        view
-        override
-        returns (uint64, uint64)
-    {
-        uint64 epoch;
-        uint64 queueSize;
-
-        if (firstAvailableEpoch < currentEpoch() + 1) {
-            epoch = currentEpoch() + 1;
-            queueSize = 0;
-        } else if (numPending >= maxChurnRate) {
-            epoch = firstAvailableEpoch + 1;
-            queueSize = 0;
+    function pushToRegistrationQueue() internal override returns (uint64, uint64) {
```

Comment:
previously i merged these into one function `nextAvailableEpoch(...)` since the logic being performed was the same - just different queues (register and exit). pros of that are: smaller contract size, reusability and consistency, cons: reduced readability which can lead to errors if the function usage isn't clear.

Which do you think is best?

Reply:
In order for them to be general to work for both queues we can't directly operate on the state I think, so I prefer to have them separate.